components/movementsensor/gpsrtkpmtk/gpsrtkpmtk.go

```diff
@@ -1,642 +0,0 @@
-		I2CAddr:     newConf.I2CAddr,
-	}
-	if config.I2CBaudRate == 0 {
-		config.I2CBaudRate = 115200
-	}
-
-	// If we have a mock I2C bus, pass that in, too. If we don't, it'll be nil and constructing the
-	// reader will create a real I2C bus instead.
-	dev, err := gpsutils.NewI2cDataReader(config, mockI2c, logger)
-	if err != nil {
-		return nil, err
-	}
-	g.cachedData = gpsutils.NewCachedData(dev, logger)
-
-	if err := g.start(); err != nil {
-		return nil, err
-	}
-	return g, g.err.Get()
-}
-
-// Start begins NTRIP receiver with i2c protocol and begins reading/updating MovementSensor measurements.
-func (g *rtkI2C) start() error {
-	g.activeBackgroundWorkers.Add(1)
-	utils.PanicCapturingGo(func() { g.receiveAndWriteI2C(g.cancelCtx) })
-
-	return g.err.Get()
-}
-
-// getStream attempts to connect to ntrip stream until successful connection or timeout.
-func (g *rtkI2C) getStream(mountPoint string, maxAttempts int) error {
-	success := false
-	attempts := 0
-
-	var rc io.ReadCloser
-	var err error
-
-	g.logger.Debug("Getting NTRIP stream")
-
-	for !success && attempts < maxAttempts {
-		select {
-		case <-g.cancelCtx.Done():
-			return errors.New("Canceled")
-		default:
-		}
-
-		rc, err = func() (io.ReadCloser, error) {
-			g.mu.Lock()
-			defer g.mu.Unlock()
-			return g.ntripClient.Client.GetStream(mountPoint)
-		}()
-		if err == nil {
-			success = true
-		}
-		attempts++
-	}
-
-	if err != nil {
-		// if the error is related to ICY, we log it as warning.
-		if strings.Contains(err.Error(), "ICY") {
-			g.logger.Warnf("Detected old HTTP protocol: %s", err)
-		} else {
-			g.logger.Errorf("Can't connect to NTRIP stream: %s", err)
-			return err
-		}
-	}
-
-	g.logger.Debug("Connected to stream")
-	g.mu.Lock()
-	defer g.mu.Unlock()
-
-	g.ntripClient.Stream = rc
-	return g.err.Get()
-}
-
-// receiveAndWriteI2C connects to NTRIP receiver and sends correction stream to the MovementSensor through I2C protocol.
-func (g *rtkI2C) receiveAndWriteI2C(ctx context.Context) {
-	defer g.activeBackgroundWorkers.Done()
-	if err := g.cancelCtx.Err(); err != nil {
-		return
-	}
-	err := g.ntripClient.Connect(g.cancelCtx, g.logger)
-	if err != nil {
-		g.err.Set(err)
-		return
-	}
-
-	if !g.ntripClient.Client.IsCasterAlive() {
-		g.logger.CInfof(ctx, "caster %s seems to be down", g.ntripClient.URL)
-	}
-
-	// establish I2C connection
-	handle, err := g.bus.OpenHandle(g.addr)
-	if err != nil {
-		g.logger.CErrorf(ctx, "can't open gps i2c %s", err)
-		g.err.Set(err)
-		return
-	}
-	defer utils.UncheckedErrorFunc(handle.Close)
-
-	// Send GLL, RMC, VTG, GGA, GSA, and GSV sentences each 1000ms
-	baudcmd := fmt.Sprintf("PMTK251,%d", g.wbaud)
-	cmd251 := movementsensor.PMTKAddChk([]byte(baudcmd))
-	cmd314 := movementsensor.PMTKAddChk([]byte("PMTK314,1,1,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0"))
-	cmd220 := movementsensor.PMTKAddChk([]byte("PMTK220,1000"))
-
-	err = handle.Write(ctx, cmd251)
-	if err != nil {
-		g.logger.CDebug(ctx, "Failed to set baud rate")
-	}
```

Comment:
Do these i2c write commands and checks exist anywhere now?

Reply:
Question: Do we need this code in the RTK at all? It exists in the gpsutils/i2c_data_reader.go. Since you got an RTK fix I assume the checks went through the data reader and in the past we were doing it twice.

Reply:
Susmita, thanks so much for noticing! You're totally right: these commands are sent when initializing the `DataReader`, called from line 135 of gpsrtk/pmtk.go, on the way to initializing `g.cachedData`. There's no need to add them in a second time. Keep the code DRY!

Reply:
we weren't resetting the pi while testing, so its possible these values were already set? haven't looked thru the code to see if the code in i2c_data_reader.go still gets called tho

Reply:
It does: `newRTKI2C()` calls `makeRTKI2C()`, which calls `gpsutils.NewI2cDataReader()`, which calls `reader.initialize()`, which sends these commands.

Reply:
going to resolve, well done noticing susmita!